collect_app/src/main/java/org/odk/collect/android/formmanagement/FormApiExceptionMapper.java

```diff
@@ -0,0 +1,23 @@
+package org.odk.collect.android.formmanagement;
+
+import android.content.res.Resources;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.openrosa.api.FormApiException;
+
+public class FormApiExceptionMapper {
+
+    private final Resources resources;
+
+    public FormApiExceptionMapper(Resources resources) {
+        this.resources = resources;
+    }
+
+    public String getMessage(FormApiException exception) {
+        if (exception.getType() == FormApiException.Type.UNKNOWN_HOST) {
+            return resources.getString(R.string.unknown_host_error, exception.getServerUrl());
```

Comment:
In this case we ALSO want to include the "if it keeps happening, talk to the person who asked you to collect data" message. I think just concatenating the two with a space would be fine (and concat does the right thing with LTR languages, I believe)